src/wallet/wallet.cpp

```diff
@@ -1462,6 +1462,34 @@ bool DummySignInput(const SigningProvider& provider, CTxIn &tx_in, const CTxOut
     return true;
 }
 
+static void FillInputToWeight(CTxIn& txin, int64_t target_weight)
+{
+    int64_t txin_weight = GetTransactionInputWeight(txin);
+    // Subtract current txin weight, which should include empty witness stack
+    int64_t add_weight = target_weight - txin_weight;
+
+    // We will want to subtract the size of the Compact Size UInt that will also be serialized.
+    // However doing so when the size is near a boundary can result in a problem where it is not
+    // possible to have a stack element size and combination to exactly equal a target.
+    // To avoid this possibility, if the weight to add is less than 10 bytes greater than
+    // a boundary, the size will be split so that 2/3rds will be in one stack element, and
+    // the remaining 1/3rd in another. Using 3rds allows us to avoid additional boundaries.
+    // 10 bytes is used because that accounts for the maximum size. This does not need to be super precise.
+    if ((add_weight >= 253 && add_weight < 263)
```

Comment:
I was confused about this as well since I had never come across compact size uints before. It helped me to read serialize.h and about tx serialization ([bip141](https://github.com/bitcoin/bips/blob/master/bip-0144.mediawiki#serialization) and [this doc](https://developer.bitcoin.org/reference/transactions.html#raw-transaction-format)).

This is just saying that when the stack size goes from 252 to 253, you need more bytes to encode the size in the serialized transaction. It's not the same as a byte boundary since (I assume) you need to reserve a few values for codifying how many bytes it is.